nextcord/ui/modal.py

```diff
@@ -89,11 +89,13 @@ def __init__(
         *,
         timeout: Optional[float] = None,
         custom_id: str = MISSING,
+        auto_defer: bool = True,
```

Comment:
I don't see the point of this since no one would want it set to False.

Responding to a modal submit is required. If you don't respond to the interaction, Discord will display an error message.

The "auto-defer" only happens if you don't respond manually to the interaction, so if it happens, it should always happen. No need to make it configurable.

Reply:
Using this attribute, you can easily make modal with something like this :
```py
class EasyModal(nextcord.ui.Modal):
  def callback(self, interaction):
    self.interaction = interaction

# you call it lile this:
modal = EasyModal(auto_defer=False)
hello_world = nextcord.ui.TextInput(label="Hello world")
modal.add_item(hello_world)
await interaction.send(modal)
await modal.wait()
# you do something with modal.interaction here
```

Reply:
It will _technically_ still be possible to do that with the defer since you can followup the interaction afterwards. (It's not great that it would require an extra API call though)

Maybe just make it never do the auto-defer? If people want to avoid the discord error, they can add the defer() to the callback themselves.

Reply:
the auto-defer is convenient for beginners and new projects, idk if we need to delete it
after it is true that it is less useful here than with a button, especially with the fact that the user can think that it does not work

Reply:
I tried that already using followups, but this leaded to some errors if you are responding outside of the callback

Reply:
Also there's no way to defer ephemerally or not, the developer can't decide about that when he's using the auto defer. Which kinda means that you aren't able to send ephemeral messages when using the EasyModal